gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/ContentsUtil.java

```diff
@@ -57,29 +59,31 @@ public static Optional<String> fetchPageContents(URL pageUrl) {
 
     private static Callable<String> fetchContentTask(URL pageUrl, ProgressIndicator indicator) {
         return () -> {
-            HttpURLConnection connection = (HttpURLConnection) pageUrl.openConnection();
-            connection.setRequestMethod("GET");
+            HttpURLConnection connection = null;
+            try {
+                connection = (HttpURLConnection) pageUrl.openConnection();
+                connection.setRequestMethod("GET");
 
-            if (connection.getResponseCode() != 200) {
-                connection.disconnect();
-                throw new ProcessCanceledException();
-            }
-
-            BufferedReader in =
-                    new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
-            StringBuilder result = new StringBuilder();
-            String inputLine;
+                if (connection.getResponseCode() != 200) {
+                    throw new ProcessCanceledException();
+                }
 
-            while ((inputLine = in.readLine()) != null) {
                 if (indicator.isCanceled()) {
                     throw new InterruptedException("Fetch cancelled");
                 }
-                result.append(inputLine);
-            }
 
-            in.close();
-            connection.disconnect();
-            return result.toString();
+                BufferedReader in =
+                        new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+
+                return in.lines().collect(Collectors.joining("\n"));
+            } catch (ConnectException e) {
+                log.error("Connection refused on page {}", pageUrl, e);
+                throw e;
```

Comment:
We just shouldn't throw the error at all. It's fine if we can't connect to an incorrect url, I'd just not report anything. Even logging I'd consider only doing it once.

Reply:
set to only debug level and not throwing anymore